### DISCUSSION TITLE
Error when accessing transcription output result

### DISCUSSION THREAD

**Discussion by dgoryeo** [SYSTRAN/faster-whisper#545] [Q&A]
I paln to swap between faster-whisper and whisper transcribe function when selected? My code looks like this: 

`result = WHISPER_MODEL.transcribe("audio.wav", **options)` 

where WHISPER_MODEL will be either faster-whisper of whisper based on user selection. 

My question: Will the output have the same format/structure?  At the moment I get an error like this when I want to access the output result:
```

TypeError                                 Traceback (most recent call last)
[<ipython-input-10-04ff465951a7>](https://localhost:8080/#) in <cell line: 146>()
    152 
    153         # check if result doesn't end with severe hallucinations
--> 154         if len(result["segments"]) == 0:
    155             break

TypeError: tuple indices must be integers or slices, not str

```

The code works fine with whisper.

**Comment by blackpolarz** [comment]
The original whisper uses something similar to a dictionary while faster-whisper uses a generator. The output is different. 
Assuming your result is the output of the transcribe, you will have to check the segments within the result and see if they exist instead of just using a len method.
